[CI] Test Failure: SearchableSnapshotsCanMatchOnCoordinatorIntegTests ... OutsideOfTheQueryRange

This was a failure on a PR build

**Build scan**:
https://gradle-enterprise.elastic.co/s/uzjhthpxcehto/console-log?task=:x-pack:plugin:searchable-snapshots:internalClusterTest

**Repro line**:
./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testSearchableSnapshotShardsAreSkippedWithoutQueryingAnyNodeWhenTheyAreOutsideOfTheQueryRange" -Dtests.seed=48922BDD4F42B52B -Dtests.security.manager=true -Dtests.locale=he-IL -Dtests.timezone=Asia/Yekaterinburg -Druntime.java=8

**Applicable branches**:
7.x

**Failure excerpt**:

```
2> java.lang.AssertionError: 	
    Expected: <0>	
         but: was <1>	
        at __randomizedtesting.SeedInfo.seed([48922BDD4F42B52B:D26CA9D3796ADCCD]:0)	
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)	
        at org.junit.Assert.assertThat(Assert.java:956)	
        at org.junit.Assert.assertThat(Assert.java:923)	
        at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testSearchableSnapshotShardsAreSkippedWithoutQueryingAnyNodeWhenTheyAreOutsideOfTheQueryRange(SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java:166)	
```

The commit it was building from would give line 166 as:
https://github.com/elastic/elasticsearch/blob/288db854c04c55de1521537c1f4e796a5c939796/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java#L166

Another one on CI for 7.11 branch: https://gradle-enterprise.elastic.co/s/rsjcmwe47dzj2

It is reproducible locally with:
> ./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testSearchableSnapshotShardsAreSkippedWithoutQueryingAnyNodeWhenTheyAreOutsideOfTheQueryRange" -Dtests.seed=1248A2E3DD7FA749 -Dtests.security.manager=true -Dtests.locale=tr-TR -Dtests.timezone=America/St_Vincent -Druntime.java=11

Failure excerpt:
```
  2> java.lang.AssertionError: 
    Expected: <0>
         but: was <1>
        at __randomizedtesting.SeedInfo.seed([1248A2E3DD7FA749:88B620EDEB57CEAF]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsCanMatchOnCoordinatorIntegTests.testSearchableSnapshotShardsAreSkippedWithoutQueryingAnyNodeWhenTheyAreOutsideOfTheQueryRange(SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java:166)
```

The failure is seed specific. So I'll leave it unmuted.

--- a/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java
@@ -103,7 +103,8 @@ public class SearchableSnapshotsCanMatchOnCoordinatorIntegTests extends BaseSear
             indexRandomDocs(indexOutsideSearchRange, between(0, 1000));
         }
 
-        int numDocsWithinRange = between(0, 1000);
+        // Index enough documents to ensure that all shards have at least some documents
+        int numDocsWithinRange = between(100, 1000);
         indexDocumentsWithTimestampWithinDate(indexWithinSearchRange, numDocsWithinRange, "2020-11-28T%02d:%02d:%02d.%09dZ");
 
         final String repositoryName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);

